src/migrations/20240116000001-add-effectivedate_for_migrated_personnel.js

```diff
@@ -0,0 +1,91 @@
+const {
+  prepMigration,
+} = require('../lib/migration');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, _Sequelize) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+      await queryInterface.sequelize.query(
+        `
+        -------------
+        -- PROCESS --
+        -------------
+        -- Get all the roles * programs * grants
+        -- Get the history of all roles, so:
+        --   roles * programs * grants * emails
+        -- Get the instances where people were set active
+        --   This is separate because they may be set active
+        --   in a different transaction than when the rest
+        --   of their data was set
+        -- Count how many times each role * program * grant was assigned
+        -- Get the latest activation timestamp for the currently assigned personnel
+        --   No one appears to have been assigned to the same role * program * grant
+        --   more than once, so this is more of a "due diligence" item
+        -- Update any blank effectiveDates with the latest activation timestamp
+
+        WITH all_assigned_roles AS (
+        SELECT DISTINCT
+          "programId" pid,
+          "grantId" grid,
+          role
+        FROM "ProgramPersonnel"
+        ),
+        all_personnel_roles AS (
+        SELECT DISTINCT
+          (new_row_data->>'programId')::int zpid,
+          (new_row_data->>'grantId')::int zgrid,
+          new_row_data->>'role' zrole,
```

Comment:
Is there concern that a record in the ZALProgramPersonel table may not have data in all four columns, as the audit log only has deltas. So if the only change was email or role the other columns might be empty.

Reply:
At least for emails, we don't seem to have records that don't also include `active:true` (I haven't checked roles).

Reply:
There's no real way to identify the relationship without all four, but it's possible to change the relationship one column at a time, which wasn't a scenario I'd considered and should be handled

Reply:
Getting this right is a pain so I instead added a protective query at the beginning that will cause a failure if the relationship values get updated in a way that would break the logic.